elixir/apps/api/test/api/client/channel_test.exs

```diff
@@ -269,7 +269,7 @@ defmodule API.Client.ChannelTest do
       channel_pid = socket.channel_pid
 
       send(channel_pid, :token_expired)
-      assert_push "disconnect", %{"reason" => "token_expired"}
+      assert_push "disconnect", %{reason: :token_expired}
```

Comment:
Previously, we had this:

```json
{
  "event": "disconnect",
  "ref": null,
  "topic": "client",
  "payload": { "reason": "token_expired" }
}
```

Reply:
ChatGPT tells me the code above serialises to the same thing?

Reply:
![image](https://github.com/firezone/firezone/assets/5486389/6a5aded6-c986-4d32-876d-257be5df7766)

Reply:
Yeah, this change is purely internal for Elixir, so it will stay as it was.

Reply:
@thomaseizinger Just for the sake of thoroughness, I suspect the reason most Elixir engineers default to using string-keyed data structures is because they won't be affected by the [Atom limit](https://www.erlang.org/doc/efficiency_guide/advanced#:~:text=By%20default%2C%20the%20maximum%20number,lowered%20using%20the%20%2Bt%20option.).

Rather than having to think about which data structures contain external data (and thus may create a DOS vector due to exhausting the Atom limit) or think about which don't, it's easier to be consistent and just use string keys everywhere.

Reply:
@jamilbk we actually use atoms because that's the maps that we are building, so there is no risk of hitting atom limit but readability is a bit better

Reply:
> Would it be possible to assert against JSON here instead? I think that would be a much more useful test because the contract between clients and portal is JSON, not elixir. Plus, it means I would be able to just copy them to write tests against them on our end :)

This is not worth the effort because we will need to throw away everything that Phoenix builds us for testing and build (and maintain) our own API testing client. The API doesn't change so often and it's pretty easy to read when you understand that the map goes into the `payload` field.

Reply:
> Yeah, this change is purely internal for Elixir

Ah okay! Thanks for clarifying. Could you extract the non-breaking changes out into a separate PR so we can keep this one focused on the replies where I have to adapt the Rust code?

Thanks! :)

Reply:
You can simply ignore changes like this, there is just a few of them:
```diff
-- %{"reason" => "token_expired"}
++ %{reason: :token_expired}
```